controllers/front/listing/SearchController.php

```diff
@@ -66,7 +66,7 @@ protected function getProductSearchQuery()
     {
         $query = new ProductSearchQuery();
         $query
-            ->setSortOrder(new SortOrder('product', Tools::getProductsOrder('by'), Tools::getProductsOrder('way')))
+            ->setSortOrder(new SortOrder('product', Tools::getProductsOrder('by', 4), Tools::getProductsOrder('way', 1)))
```

Comment:
We had a look yesterday at all these PRs regarding the sort orders, and your seem to fix an issue.
However, if you look at the method `Tools::getProductsOrder()`, you could see that you force the values here.

This line is the equivalent of 
```php
->setSortOrder(new SortOrder('product', 'position', 'desc'))
```

which is quite easier to read, don't you think?

Reply:
@Quetzacoalt91 are you sure that if you hardcode these values you can still change the sorting way and sorting field?

Reply:
@Quetzacoalt91 you are right :  
 Tools::getProductsOrder('by', 4)  means  'position'
 Tools::getProductsOrder('way', 1) means 'desc'
You version of the fix is the same as mine but simpler.

@kpodemski, the only one place where the customer can change the sorting way of the search is on the search page after clicking the search button. I tested it - it is working.

@Quetzacoalt91 ,  should I change the code in my branch to your version or you will do it?

Reply:
@sbordun yes I think you should :)

Reply:
@kpodemski I can confirm again this later, this is what I checked on the first try. :)